### DISCUSSION TITLE
How to Customize Drag Behavior and Make Indicator Occupy Full Space in GrapesJS

### DISCUSSION THREAD

**Discussion by fredrickreuben** [GrapesJS/grapesjs#6154] [General]
I’m currently working on customizing the drag-and-drop behavior in GrapesJS, and I need help with a few things related to the default behavior:

1. Static Green Line: By default, a green line appears to indicate the drop position during the drag action. I’d like to either remove or customize this line.

2. Full-Space Indicator: Instead of just a line, I want the drag indicator to occupy the entire space (e.g., filling the area between the elements where the dragged component is about to be dropped). I’d like to change the appearance and behavior of this indicator.

So far, I’ve tried inspecting the elements, but I haven’t been able to identify the correct CSS selectors or classes to target for these changes. Could someone guide me on how to:

1. Remove or modify the default green line.
2. Make the indicator fill the available space instead of just a line.
3. Apply custom styles to the element during dragging (e.g., highlighting, background colors, borders, etc.).

If there’s a configuration option or specific code snippet that can help achieve this, I’d greatly appreciate it.

**Comment by mohamedsalem401** [comment]
@fredrickreuben 
The placeholder class name goes as follows:
* **Base class:** `.{prefix}-placeholder`, where `.{prefix}` is the prefix defined for your GrapesJS instance. For example, if your prefix is `gjs`, the base class would be `.gjs-placeholder`.
* **Orientation classes:** To specify the orientation of the placeholder, you can add either `.horizontal` or `.vertical` as a modifier class. For example, a horizontal placeholder would have the class `.gjs-placeholder.horizontal`.

There's also an internal div, here's the actual code
```ts
<div class="${ppfx}placeholder">
  <div class="${ppfx}placeholder-int"></div>
</div>
```

It may not be a great solution to override the CSS for the placeholder, but I think it should work though

**Reply by fredrickreuben** [reply]
I solved it using Canvas Spot thanks for your answer

**Reply by Turzapodder** [reply]
Hey Hi, can you please give me suggestion how can I achieve that? I tried with canvas spot but didn't work. A Code snippet will be very helpful. Another thing , is it working smoothly?

**Reply by Abhinav-birdeye** [reply]
Did anyone get an answer to this yet? How does one use canvas spot to replace the drop indicators?

**Comment by hamzaamirmukhtar** [comment]
Hi! This line (placeholder) can be edited by using classes as follows: 

```
:root {
  --gjs-placeholder-background-color: theme("your-colour") !important;
}
```
```
.gjs-placeholder.horizontal {
  border-color: rgba(0, 0, 0, 0) var(--gjs-placeholder-background-color) !important;
}
```
```
.gjs-placeholder.vertical {
  border-color: rgba(0, 0, 0, 0) var(--gjs-placeholder-background-color) !important;
}
```

**Comment by Abhinav-birdeye** [comment]
Still waiting for code-snippets/help on this. @artf , can this be done using Canvas Spots?
